src/parser/jobs/pld/modules/Goring.tsx

```diff
@@ -1,62 +1,300 @@
+		this.pushApplication(applicationKey, statusId, event, clip)
+		if (clip > MAX_ALLOWED_CLIPPING) {
+			this.trackedClipping = true
+		}
+		trackerInstance[statusId].lastApplication = event.timestamp
+	}
+
+	// calculating separately in case we ever decide to split the outputs
+	private getGoringUptime() {
+		const statusTime = this.statuses.getUptime(this.data.statuses.GORING_BLADE, this.actors.foes)
+		const uptime = this.parser.currentDuration - this.invulnerability.getDuration({types: ['invulnerable']})
+		return (statusTime / uptime) * 100
+	}
+
+	private getValorUptime() {
+		const statusTime = this.statuses.getUptime(this.data.statuses.BLADE_OF_VALOR, this.actors.foes)
+		const uptime = this.parser.currentDuration - this.invulnerability.getDuration({types: ['invulnerable']})
+		return (statusTime / uptime) * 100
+	}
+
+	private onComplete() {
 		this.checklist.add(new Rule({
-			name: 'Keep your Goring Blade up',
-			description: <Trans id="pld.goring.checklist.goringblade.description">
-				As a Paladin, <ActionLink {...ACTIONS.GORING_BLADE}/> is a significant portion of your sustained
-				damage, and is required to kept up for as much as possible, for the best damage output.
+			name: <Trans id="pld.goring.checklist.dots.name">Keep your DoTs up</Trans>,
+			description: <Trans id="pld.goring.checklist.dots.description">
+				As a Paladin, <DataLink action="GORING_BLADE" /> and <DataLink action="BLADE_OF_VALOR" /> are a significant portion of your sustained damage, and are required to be kept up for as much as possible for the best damage output.
 			</Trans>,
-			target: 90,
+			target: 95,
 			requirements: [
 				new Requirement({
-					name: <Trans id="pld.goring.checklist.goringblade.requirement.uptime"><ActionLink {...ACTIONS.GORING_BLADE} /> uptime</Trans>,
-					percent: () => this.getUptimePercent(STATUSES.GORING_BLADE.id),
+					name: <Trans id="pld.goring.checklist.dots.requirement.name">Combined DoT uptime</Trans>,
+					percent: () => this.getGoringUptime() + this.getValorUptime(),
 				}),
 			],
 		}))
 	}
 
-	override addClippingSuggestions() {
-		const goringClipPerMinute = this.getClippingAmount(this.data.statuses.GORING_BLADE.id)
+	private createTargetStatusTable(target: DotTargetData) {
+		let totalMajorDotClip = 0
 
-		// Suggestion for Goring Blade DoT clipping
-		this.suggestions.add(new TieredSuggestion({
-			icon: ACTIONS.GORING_BLADE.icon,
-			content: <Trans id="pld.goring.suggestions.goringblade.content">
-				Avoid refreshing <ActionLink {...ACTIONS.GORING_BLADE} /> significantly before it's expiration.
-			</Trans>,
-			why: <Trans id="pld.goring.suggestions.goringblade.why">
-				An average of {this.parser.formatDuration(goringClipPerMinute, 1)} seconds of <StatusLink {...STATUSES.GORING_BLADE}/> per minute lost to early refreshes.
-			</Trans>,
-			tiers: SEVERITIES.CLIPPING,
-			value: goringClipPerMinute,
-		}))
+		let combinedDotStatuses: DotApplicationData[]
```

Comment:
Kinda awkward here, you can avoid the null checks by optimistically concat/summing/whatever this. Since it initialises to zero, you can safely add goring and valor without checking if they're viable I think.